### PR TITLE
Same linter settings for vs code and ci worfklow

### PR DESCRIPTION
**Pull request by mml5bg**
# Description

Making linter configuration the same for VSCode and CI workflow.

### Current setup:
In CI workflow [public pre commit action](https://github.com/pre-commit/action) is used, and the settings are different because of different `.pre-commi-config.yaml` files what is leading to different behavior in VS Code and CI Workflow.

### Approach implemented in this PR:
Using the same `.pre-commit-config.yaml` for all executions (no matter is it executing in VS Code or CI Workflow).
It is implemented creating the custom pre-commit action which uses `.pre-commit-config.yaml` from the root of the repo, and the same file is used by VS Code extenstion for pre-commit which will execute pre-commit on save and developer will be immediately notified about issues. And on the end, this pre-commit is preventing commits if checks are not successfull.   

## Azure DevOps PBI/Task reference

[AB#10473](https://softwaredefinedvehicle.visualstudio.com/SoftwareDefinedVehicle/_workitems/edit/10473)

## Checklist

* [ ] Vehicle App can be started with dapr run and is connecting to vehicle data broker
* [ ] Vehicle App can process MQTT messages and call the seat service
* [ ] Vehicle App can be deployed to local K3D and is running
* [ ] Created/updated tests, if necessary. Code Coverage percentage on new code shall be >= 70%.
* [ ] Extended the documentation in Velocitas repo
* [ ] Extended the documentation in README.md

* [x] Devcontainer can be opened successfully
* [x] Devcontainer can be opened successfully behind a corporate proxy
* [x] Devcontainer can be re-built successfully

* [ ] Release workflow is passing

### CODE REVIEW

**Comment by SebastianSchildt** [comment]
The problem is the automated commit 
image

Clean solutions
 - Do no use GH actions that change code/do commits. Bad style
 - If you want Linter/license checks just treat it as "check" and when it errors, go fix your code and make a new commit

Semi-clean solutions:
 - Fix your local branch, and force push

(Fast) solution:
 - Convince your committer(s)  to merge anyway
 (this is fast for you, but not the comitter, because he/she now needs to _manually_ check every commit, making sure all "non-automated"/non-trivial ones are signed off correctly)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -152,14 +155,6 @@ jobs:
     runs-on: ubuntu-latest
     needs: build-image-amd64
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Clone Release Documentation Action repository
```
Comment (DocMoebiuz) [inline .github/workflows/ci.yml:157]: Why did the order change of this task?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -34,16 +34,10 @@ jobs:
 
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
```
Comment (DocMoebiuz) [inline .github/workflows/ci.yml:37]: what exactly is the difference between v2 and v3?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -34,16 +34,10 @@ jobs:
...
+        uses: actions/checkout@v3
         with:
           ref: ${{ github.head_ref }}
 
-      - name: Clone Release Documentation Action repository
```
Comment (DocMoebiuz) [inline .github/workflows/ci.yml:41]: why did the order of this step change? it is now executed later.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -57,7 +51,13 @@ jobs:
           pip install -r test/requirements.txt
 
       - name: Run Linters
-        uses: pre-commit/action@v3.0.0
```
Comment (DocMoebiuz) [inline .github/workflows/ci.yml:60]: Where did the old pre-commit action gets its settings from?

**Review by DocMoebiuz — CHANGES_REQUESTED** [review]
Please add more information in the PR itself to help understand what the PR changed and what new concepts have been introduced.

From my understanding the PR takes care of:
* Making sure that the pre-commit hook when running locally and on GitHub in the cloud are using the same settings. (Can you please explain a bit more why this was not the case until now?)
* The execution of the pre-commit locally was not intuitive and is now performed automatically by VSCode on each save. 
  * Is this really convenient? I  have to try first 
  * Why is the pre-commit hook not executed by my git client BEFORE COMMITING changes in my local clone? It should fail locally and prevent me from committing changes. At the latest, when pushing.... but I have to read up on the pre-commit concept again.
  
  
The original request is still open. "Same settings for VSCode and CI Workflow".

**Comment by mml5bg** [comment]
> Please add more information in the PR itself to help understand what the PR changed and what new concepts have been introduced.
> 
> From my understanding the PR takes care of:
> 
>     * Making sure that the pre-commit hook when running locally and on GitHub in the cloud are using the same settings. (Can you please explain a bit more why this was not the case until now?)
> 
>     * The execution of the pre-commit locally was not intuitive and is now performed automatically by VSCode on each save.
>       
>       * Is this really convenient? I  have to try first
>       * Why is the pre-commit hook not executed by my git client BEFORE COMMITING changes in my local clone? It should fail locally and prevent me from committing changes. At the latest, when pushing.... but I have to read up on the pre-commit concept again.
> 
> 
> The original request is still open. "Same settings for VSCode and CI Workflow".
PR description is updated with approach and explanation.

Of course, you should try.
Now, it is executed before commiting and it is preventing to commit code before all issues are resolved. I already tested it.

And on the end. Original request is satisfied because, now configuration is exactly the same for VS Code and CI Workflow. How? Because, they are all using the same tool (pre-commit) and that tool uses the only one configuration file whenever it is executed (locally in VS Code or in CI Workflow). Now is not possible to have situation where someone commit something and CI workflow fails because of the linter issues, because of new setup, implemented in this PR.

**Review by BjoernAtBosch — COMMENTED** [review]
Looks fine for me

**Review by BjoernAtBosch — APPROVED** [review]

**Review by DocMoebiuz — APPROVED** [review]
Do we have impact on our documentation repository?
I think we should mention something about this feature if we haven't yet.

**Pull request merged**